ApeWorX/ape
# feat: wait for transaction confirmations

### What I did

fixes: #274 

### How I did it

Add parameter to Transaction for setting the required confirmations
Add default required confirmations to the `NetworkAPI`
If transaction was not given a `required_confirmations` kwarg, use the default from the `NetworkAPI`

### How to verify it

* Run transactions on development - should be no confirmations needed
* Run transactions on testnets - should wait for 1 transaction
* Try adjusting the required confirmations on transfers, deploys, and contract txns

### Checklist
